### DISCUSSION TITLE
[Misleading title and description] There seems to be an issue with renovate, its merging before commits are built on cicd.

### DISCUSSION THREAD

**Discussion by Ranjith072** [renovatebot/renovate#8453] [Request Help]
**What Renovate type, platform and version are you using?**

Hosted app

**Describe the bug**
renovate is merging updates before they are built on CICD pipeline because of which all the builds are failing and also for cdk it is updating partially(updating package.json but missed to update setup.py).

**Relevant debug logs**

![image](https://user-images.githubusercontent.com/57263575/106176453-9accc880-6197-11eb-999c-974f3ee4f290.png)

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [x] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

**Comment by viceice** [comment]
I think this happened because of github actions failure, see https://www.githubstatus.com/incidents/tf9v5jjmq2lg

@Ranjith072 Are you using [stabilityDays](https://docs.renovatebot.com/configuration-options/#stabilitydays)?

**Comment by Ranjith072** [comment]
yes allowed to merge only after 3 days.

**Comment by Ranjith072** [comment]
{
  "extends": [
    "config:base"
  ],
  "automergeType": "branch",
  "automerge": true,
  "masterIssue": true,
  "major": {
    "automerge": false
  },
  "pin": {
    "automerge": true
  },
  "pinDigests": true,
  "packageRules": [
    {
      "packageNames": ["awscli", "aws-sdk", "boto3", "boto3-stubs", "botocore"],
      "extends": ["schedule:monthly"],
      "stabilityDays": 0,
      "prNotPendingHours": 0,
      "groupName": "aws packages"
    },
    {
      "packagePatterns": [
        "^org\\.jetbrains\\.kotlinx:"
      ],
      "versionScheme": "semver"
    }
  ],
  "lockFileMaintenance": {
    "enabled": true
  },
  "ignorePaths": [
      "sql-server-dependency/Dockerfile"
  ],
  "stabilityDays": 3,
  "prNotPendingHours": 74,
  "timezone": "Europe/Oslo",
  "semanticCommits": true
}

**Comment by viceice** [comment]
Please format your comment

Did you check the pr /commit? Does it have a succeed status check?

**Comment by rarkins** [comment]
You need to add branch protection identifying mandatory status checks for the repo to prevent cases like today where GitHub actions simply didn't run.

**Comment by Ranjith072** [comment]
i do have that in place.
![image](https://user-images.githubusercontent.com/57263575/106179567-7ffc5300-619b-11eb-8405-b41bfee745aa.png)

**Comment by rarkins** [comment]
But in that case GitHub should be blocking Renovate from merging. It should be impossible for a renovate to merge if you had those settings and it's a GitHub bug if so.

**Comment by rarkins** [comment]
Please provide debug logs for that branch for the job when it was automerged. The screenshot you provided instead of logs doesn't help any.

**Comment by Ranjith072** [comment]
please find the debug logs attached.
[debug_logs.txt](https://github.com/renovatebot/renovate/files/5889361/debug_logs.txt)

**Comment by rarkins** [comment]
And did you see this?

```
DEBUG: branch status check result(branch="renovate/pytest-6.x")
{
  "state": "success",
  "statuses": [
    {
      "url": "https://api.github.com/repos/westernbulk/western-ava/statuses/28f8c79e03ad56467be4b5301b75ffa739b5924d",
      "avatar_url": "https://avatars.githubusercontent.com/u/52701652?v=4",
      "id": 11996866831,
      "node_id": "MDEzOlN0YXR1c0NvbnRleHQxMTk5Njg2NjgzMQ==",
      "state": "success",
      "description": "This commit looks good",
      "target_url": "https://jenkins.devops.westernbulk.cloud/job/western-ava/job/renovate%252Fpytest-6.x/23/display/redirect",
      "context": "continuous-integration/jenkins/branch",
      "created_at": "2021-01-28T13:57:17Z",
      "updated_at": "2021-01-28T13:57:17Z"
    },
    {
      "url": "https://api.github.com/repos/westernbulk/western-ava/statuses/28f8c79e03ad56467be4b5301b75ffa739b5924d",
      "avatar_url": "https://avatars.githubusercontent.com/in/2740?v=4",
      "id": 11998351783,
      "node_id": "MDEzOlN0YXR1c0NvbnRleHQxMTk5ODM1MTc4Mw==",
      "state": "success",
      "description": "Updates have met stability days requirement",
      "target_url": "https://docs.renovatebot.com/",
      "context": "renovate/stability-days",
      "created_at": "2021-01-28T15:51:29Z",
      "updated_at": "2021-01-28T15:51:29Z"
    }
  ]
}
```

**Reply by rarkins** [reply]
Conclusion: Both mandatory status checks reported `success`, branch was merged - nothing wrong
